pkg/download/downloader.go

```diff
@@ -61,20 +67,53 @@ func Download(downloader Downloader) (int, io.ReadCloser, error) {
 		return response.StatusCode, response.Body, nil
 	}
 
-	err = fmt.Errorf("Status code %d while downloading blob '%s'. Use either a public script URI that points to .sh file, Azure storage blob SAS URI or storage blob accessible by a managed identity and retry. For more info, refer https://aka.ms/RunCommandManagedLinux", response.StatusCode, request.URL.Opaque)
+	errString := ""
+	requestId := response.Header.Get(xMsServiceRequestIdHeaderName)
 	switch downloader.(type) {
 	case *blobWithMsiToken:
 		switch response.StatusCode {
 		case http.StatusNotFound:
-			notFoundError := fmt.Errorf("Make sure Azure blob '%s' and managed identity exist, and identity has been given access to storage blob's container with 'Storage Blob Data Reader' role assignment. In case of user assigned identity, make sure you add it under VM's identity. For more info, refer https://aka.ms/RunCommandManagedLinux", request.URL.Opaque)
-			return response.StatusCode, nil, errors.Wrapf(notFoundError, MsiDownload404ErrorString)
-		case http.StatusForbidden:
+			notFoundError := fmt.Sprintf("RunCommand failed to download blob '%s' and recieved response code '%s'. Make sure Azure blob and managed identity exist, and identity has access to storage blob's container with 'Storage Blob Data Reader' role assignment. For user assigned identity, add it under VM's identity. For more info, see https://aka.ms/RunCommandManagedLinux", request.URL.Opaque, response.Status)
```

Comment:
nit: received

Reply:
There are a lot of grammar mistakes here and in the other errors. I recommend running them through ChatGPT (Please correct my grammar. {text}) to correct them.